Shared/Samples/Edit feature attachments/EditFeatureAttachmentsView.Model.swift

```diff
@@ -0,0 +1,145 @@
+        /// A Boolean value that indicates whether the callout placement should be offset for the map magnifier.
+        @Published var calloutShouldOffset = false
+        
+        /// The `FeatureLayer` populated with data by the feature table using the remote service url. 
+        var featureLayer: FeatureLayer = {
+            let featureTable = ServiceFeatureTable(url: .featureServiceURL)
+            var featureLayer = FeatureLayer(featureTable: featureTable)
+            return featureLayer
+        }()
+        
+        init() {
+            map.addOperationalLayer(featureLayer)
+        }
+        
+        /// Updates the location of the callout placement to a given screen point.
+        /// - Parameter location: The screen point at which to place the callout.
+        func updateCalloutPlacement(to location: Point) {
+            // Create an offset to offset the callout if needed, e.g. the magnifier is showing.
+            let offset = calloutShouldOffset ? CGPoint(x: 0, y: -70) : .zero
+            calloutPlacement = .location(location, offset: offset)
+        }
+        
+        /// Selects the feature that is tapped.
+        /// - Parameter feature: The feature to set as the selected feature.
+        func setSelectedFeature(for feature: ArcGISFeature) async throws {
+            selectedFeature = feature
+            featureLayer.selectFeature(selectedFeature!)
+            try await fetchAttachmentsAndUpdateFeature()
+        }
+        
+        /// Updates the callout text for the selected feature.
+        private func updateCalloutDetailsForSelectedFeature() {
+            if let selectedFeature = selectedFeature {
+                let title = selectedFeature.attributes["typdamage"] as? String
+                calloutText = title ?? "Callout"
+                calloutDetailText = "Number of attachments: \(attachments.count)"
+            }
+        }
+        
+        /// Adds a new attachment with the given parameters and synch this change with the server.
+        /// - Parameters:
+        ///   - named: The attachment name.
+        ///   - type: The attachments data type.
+        ///   - dataElement: The attachment data.
+        func addAttachment(named: String, type: String, dataElement: Data) async throws {
+            if let table = selectedFeature?.table as? ServiceFeatureTable, table.hasAttachments,
+               let feature = selectedFeature {
+                _ = try await feature.addAttachment(
+                    named: "Attachment.png",
+                    contentType: "png",
+                    data: dataElement
+                )
+                try await syncChanges()
+                try await fetchAttachmentsAndUpdateFeature()
+            }
+        }
+        
+        /// Deletes the specified attachment and syncs the changes with the server.
+        /// - Parameter attachment: The attachment to be deleted.
+        func deleteAttachment(attachment: Attachment) async throws {
```

Comment:
```suggestion
        func deleteAttachment(_ attachment: Attachment) async throws {
```
`attachment` should have an anonymous argument label so that type information is not repeated at the call site.